node_io/lib/src/streams.dart

```diff
@@ -204,4 +205,13 @@ class NodeIOSink extends WritableStream<List<int>> implements IOSink {
   void writeln([Object obj = ""]) {
     _write(encoding.encode("$obj\n"));
   }
+
+  @override
+  add(List<int> data) {
+    // Add as buffer
+    if (data is! Uint8List) {
+      data = new Uint8List.fromList(data);
+    }
```

Comment:
This shouldn't be needed since NodeIOSink converts everything to Node.js `Buffer` before writing (see `convert` function in `NodeIOSink` constructor). Curious if it didn't work for you in some cases?

Reply:
Actually the test add_bytes in file_test fails without this.

And indeed the [Buffer doc](https://nodejs.org/api/buffer.html#buffer_buffer_from_buffer_alloc_and_buffer_allocunsafe) states that it works as we expect if the first argument is an ArrayBuffer (hence the need of a Uint8List) not just an int array

Reply:
OK, makes sense.